connectivity/drivers/cellular/TELIT/ME310/TELIT_ME310_CellularStack.cpp

```diff
@@ -440,7 +440,7 @@ nsapi_size_or_error_t TELIT_ME310_CellularStack::socket_recvfrom_impl(CellularSo
                 // read() should not fail
                 success = false;
             }
-        } else if (timer.read_ms() < ME310_SOCKET_TIMEOUT) {
+        } else if (std::chrono::duration_cast<std::chrono::milliseconds>(timer.elapsed_time()) < socket_timeout) {
```

Comment:
The duration cast is not required here. You only need it when doing a precision-losing conversion.

Comparisons are always exact, regardless of the precision on either side. The less-precise thing gets converted to the more-precise thing.

```suggestion
        } else if (timer.elapsed_time() < socket_timeout) {
```

Reply:
i changed it accordingly to your suggestions